File: make_str_ids.pl
#!/usr/bin/perl -l

use strict;
use warnings;

my $header = <STDIN>;
chomp $header;
my @HF = split /\t/, $header;
my @id_mask = map { ($_ =~ m!ID$!) ? 1 : 0} @HF;
print $header;


while(<STDIN>) {
  chomp;
  
  my @F = split /\t/;
  print join "\t", map {if ($id_mask[$_]) { $_ = sprintf("%09d", $F[$_]) } else { $_ = $F[$_] } $_ }  0..$#F;  
}

File: make_str_ids.py
#!/usr/bin/env python
# -*- coding:utf8 -*-

import csv
import sys
import re
if __name__=="__main__":
  reader = csv.reader(sys.stdin, delimiter='\t', quotechar='"')
  writer = csv.writer(sys.stdout, delimiter='\t', quotechar='"')
  header = reader.next()
  print '\t'.join(header)
  id_mask = [1 if re.search(r'ID$', f) else 0 for f in header]
  for row in reader:
    row = ["%09u"%int(row[i]) if id_mask[i] else row[i] for i in range(len(row)) ]
    #print '\t'.join(row)
    writer.writerow(row)
